type_info_test.go

```diff
@@ -226,6 +251,10 @@ func TestErrTypeInfo(t *testing.T) {
 	nilStructEntry, err := NewStructEntry(nil, "hello")
 	require.NoError(t, err)
 
+	// Invalid ARRAY entry.
+	_, err = NewArrayInfo(validInfo, 0)
```

Comment:
We do not allow arrays of zero length? I know, this would make no sense in the real world but should we block this even if it is valid (at least in Go - no idea how DuckDB behaves).